hera_cal/datacontainer.py

```diff
@@ -498,3 +498,53 @@ def select_or_expand_times(self, new_times, in_place=True):
 
         if not in_place:
             return dc
+
+
+class RedDataContainer(DataContainer):
+
+    def __init__(self, data, reds=None, antpos=None, bl_error_tol=1.0):
+        '''Structure for containing redundant visibilities that can be accessed by any
+        one of the redundant baseline keys (or their conjugate).
+
+        Arguments:
+            data: DataContainer or dictionary of visibilities, just as one would pass into DataContainer().
+                Will error if multiple baselines are part of the same redundant group.
+            reds: list of lists of redundant baseline tuples, e.g. (ind1, ind2, pol)
+            antpos: dictionary of antenna positions in the form {ant_index: np.array([x, y, z])}.
+                Will error if one tries to provide both reds and antpos. If neither is provided,
+                will try to to use data.antpos (which it might have if its is a DataContainer).
+            bl_error_tol: the largest allowable difference between baselines in a redundant group
+                (in the same units as antpos). Normally, this is up to 4x the largest antenna position
+                error. Will only be used if reds is inferred from antpos.
+        '''
+        if reds is not None and antpos is not None:
+            raise ValueError('Can only provide reds or antpos, not both.')
+
+        super().__init__(data)
+
+        # Figure out reds
+        if reds is not None:
+            self.reds = reds
+        else:
+            from .redcal import get_reds
+            if antpos is not None:
+                self.reds = get_reds(antpos, pols=self.pols(), bl_error_tol=bl_error_tol)
+            elif hasattr(self, 'antpos') and self.antpos is not None:
+                self.reds = get_reds(self.antpos, pols=self.pols(), bl_error_tol=bl_error_tol)
+            else:
+                raise ValueError('Must provide reds, antpos, or have antpos available at data.antpos')
+
```

Comment:
Are we sure that the right way to do this is to make new keys, or do we want to map getitem to the appropriate baseline before handing it off to the super?

Reply:
This is a good idea. I'll do that.

Reply:
I'm still confused about the way you did it vs overwritting __getitem__() from DataContainer. Maybe we can discuss.

Reply:
I did overwrite `__getitem__`

Reply:
Sorry, I think I was viewing a cached version of the PR from before vacation.

Reply:
No problem.